src/object/class_object.c

```diff
@@ -4103,9 +4103,23 @@ classobj_find_constraint_by_attrs (SM_CLASS_CONSTRAINT * cons_list, DB_CONSTRAIN
 
 	      if (i == len)
 		{
-		  return cons;	/* match */
+		  if (filter_predicate)
+		    {
+		      if (!cons->filter_predicate)
+			{
+			  continue;
+			}
+
+		      if (strcmp (filter_predicate->pred_string, cons->filter_predicate->pred_string))
+			{
+			  continue;
+			}
```

Comment:
NULL check for pred_string is required.

Reply:
the "pred_string" is set only when the filter_predicate is set, therefore pred_string can not be null in this case.
but I put the code to avoid compare with null string.